frontend/src/features/splReports/containers/SplReportContainer.tsx

```diff
@@ -123,54 +171,60 @@ const SplReportContainer: React.FunctionComponent<ISplReportContainerProps> = ()
   };
 
   return (
-    <SplReportLayout
-      showSidebar={showSidebar}
-      setShowSidebar={setShowSidebar}
-      onOpen={report => setCurrentReport(report)}
-      onDelete={onDelete}
-      onFinal={onFinal}
-      onRefresh={refreshSnapshots}
-      onSave={onSave}
-      onAdd={addReport}
-      onExport={exportReport}
-      onFromChange={(id: number) => {
-        changeFrom({ ...currentReport, from: _.find(reports, { id: id })?.to } as IReport);
-      }}
-      reports={reports}
-      snapshots={snapshots}
-      currentReport={currentReport}
+    <SplReportContainerContext.Provider
+      value={{ snapshotFilter: snapshotFilter, setSnapshotFilter }}
     >
-      <GenericModal
-        display={!!reportIdToDelete}
-        title="Confirm Delete"
-        okButtonText="Delete"
-        okButtonVariant="danger"
-        cancelButtonText="Cancel"
-        cancelButtonVariant="secondary"
-        message="Are you sure you want to delete this report? this action cannot be undone."
-        handleOk={() => {
-          const report = _.find(reports, { id: reportIdToDelete });
-          report && deleteReport(report);
-          setReportIdToDelete(undefined);
+      <SplReportLayout
+        showSidebar={showSidebar}
+        setShowSidebar={setShowSidebar}
+        onOpen={report => setCurrentReport(report)}
+        onDelete={onDelete}
+        onFinal={onFinal}
+        onRefresh={refreshSnapshots}
+        onSave={onSave}
+        onAdd={addReport}
+        onExport={exportReport}
+        onFromChange={(id: number) => {
+          changeFrom({ ...currentReport, from: _.find(reports, { id: id })?.to } as IReport);
```

Comment:
Do we need lodash anymore?  Seems like array's already have most of this functionality.